chain/chain/src/resharding/manager.rs

```diff
@@ -23,15 +27,27 @@ pub struct ReshardingManager {
     /// A handle that allows the main process to interrupt resharding if needed.
     /// This typically happens when the main process is interrupted.
     pub resharding_handle: ReshardingHandle,
+    /// Takes care of performing resharding on the flat storage.
+    pub flat_storage_resharder: Option<FlatStorageResharder>,
 }
 
 impl ReshardingManager {
     pub fn new(
         store: Store,
         epoch_manager: Arc<dyn EpochManagerAdapter>,
+        runtime_adapter: Arc<dyn RuntimeAdapter>,
```

Comment:
Just for my education, why do you need the runtime adapter in this PR?

Reply:
The reason is that now `FlatStorageResharder` is owned and created by `ReshardingManager`. `runtime` is needed by `FlatStorageResharder`